src/tokens/formats/json-formats.js

```diff
@@ -0,0 +1,70 @@
+const jsonVariables = {
+  name: 'rvt/json/variables',
+  formatter: function(dictionary) {
+    const allProperties = dictionary.allProperties;
+    const properties = dictionary.properties;
+    // eslint-disable-next-line prefer-const
+    let categories = Object.keys(properties);
+
+    // Wrapper is the string that will be built into the final output
+    let wrapper = ``;
+    wrapper += `{\n`;
+
+    // Output holds the output from each category loop
+    let output = ``;
+    categories.forEach(function(item) {
+      output += `  ${JSON.stringify(item)}:[\n`;
+
+      // Set counter to check for first item in output loop
+      let counter = 0;
+
+      // Loop through allProperties to find entries which match the category
+      for (const key in Object.entries(allProperties)) {
+        const category = allProperties[key].attributes.category;
+        const value = allProperties[key].value;
+
+        // The path is the fully-articulated property name as an array
+        // Combine the array (kebab-case) or output item if single item
+        const path =
+          allProperties[key].path.length > 1
+            ? allProperties[key].path.join('-')
+            : allProperties[key].path[0];
+
+        // Check that the category matches the item
+        if (category === item) {
+          // Increase counter
+          counter += 1;
+
+          /**
+           * Build json output string for item
+           *
+           *  {
+           *      "name": "fully-articulated property name",
+           *      "value": "value"
+           *  }
+           */
+
+          let line = ``;
+
+          // If first item, do not add initial comma
+          line += counter === 1 ? `    {\n` : `,\n    {\n`;
```

Comment:
If you've passed in `index` as the second argument to you `forEach` callback, you could check that it's the first item in the array. Similar to your counter variable, if the index is `0` (first item in the array), then you don't add the comma.

```suggestion
          line += (index === 0) ? `    {\n` : `,\n    {\n`;
```

I think what you're doing here works, but since you get a counter for free with `forEach`, it might be a good opportunity to clean it up a bit, if I'm understanding things correctly. **I haven't tested this code**, but I _think_ it should work.